File: backend/app/schemas/capability_recommendation.py
from typing import Literal

from pydantic import BaseModel, Field


CapabilityRecommendationType = Literal["mcp", "skill"]


class CapabilityRecommendationRequest(BaseModel):
    query: str = Field(default="", max_length=1000)
    limit: int = Field(default=3, ge=1, le=10)


class CapabilityRecommendationItem(BaseModel):
    type: CapabilityRecommendationType
    id: int
    name: str
    description: str | None = None
    score: float
    default_enabled: bool


class CapabilityRecommendationResponse(BaseModel):
    query: str
    items: list[CapabilityRecommendationItem] = Field(default_factory=list)

File: frontend/features/task-composer/types/capability-recommendation.ts
export type CapabilityRecommendationType = "mcp" | "skill";

export interface CapabilityRecommendation {
  type: CapabilityRecommendationType;
  id: number;
  name: string;
  description: string | null;
  score: number;
  default_enabled: boolean;
}

export interface CapabilityRecommendationsResponse {
  query: string;
  items: CapabilityRecommendation[];
}
